src/test/java/com/cloudbees/jenkins/support/CheckFilterTest.java

```diff
@@ -0,0 +1,412 @@
+
+        //project.getBuildersList().add(new SimpleBuilder());
+        QueueTaskFuture<FreeStyleBuild> buildQueued = project.scheduleBuild2(0);
+        return buildQueued;
+    }
+
+    private void assertComponent(List<Class<? extends Component>> componentClasses, FileChecker checker) throws IOException {
+        File fileZip = new File(temp.getRoot(), "filteredBundle.zip");
+        Files.deleteIfExists(fileZip.toPath());
+
+        try (FileOutputStream zipOutputStream = new FileOutputStream(fileZip)) {
+            ExtensionList<Component> existingComponents = ExtensionList.lookup(Component.class);
+            List<Component> componentsRequested = existingComponents.stream().filter(existingComponent -> componentClasses.contains(existingComponent.getClass())).collect(Collectors.toList());
+            SupportPlugin.writeBundle(zipOutputStream, componentsRequested);
+
+            //ZipArchiveInputStream zip = new ZipArchiveInputStream(new FileInputStream(fileZip));
+            try(ZipFile zip = new ZipFile(fileZip)) {
+                Enumeration<ZipArchiveEntry> zipFileEntries = zip.getEntries();
+
+                while (zipFileEntries.hasMoreElements()) {
+                    ZipArchiveEntry entry = zipFileEntries.nextElement();
+                    if (entry.isDirectory()) {
+                        break;
+                    }
+                    if (entry.getSize() > 2048 * 1024) {
+                        LOGGER.log(Level.WARNING, "Cannot check file '%s' because its content is bigger than 2Mb", entry.getName());
+                        break;
+                    }
+
+                    String content = getContentFromEntry(zip, entry);
+                    checker.check(entry.getName(), content);
+                }
+            }
+        }
+    }
+
+    private String getContentFromEntry(ZipFile zip, ZipArchiveEntry entry) throws IOException {
+        if (entry.getSize() == 0) {
+            return "";
+        }
+
+        String content;
+        try(ByteArrayOutputStream out = new ByteArrayOutputStream();
+            BufferedInputStream is = new BufferedInputStream(zip.getInputStream(entry))) {
+
+            int currentByte;
+            // buffer of same size as the entry
+            byte data[] = new byte[(int) entry.getSize()];
+
+            // read and write until last byte is encountered
+            while ((currentByte = is.read(data, 0, (int) entry.getSize())) != -1) {
+                out.write(data, 0, currentByte);
+            }
+            out.flush();
+            content = new String(out.toByteArray(), Charset.forName("UTF-8"));
+        }
+        return content;
+    }
+
+    private static class FileChecker {
+        private Set<FileToCheck> fileSet = new HashSet<>();
+        private Set<String> words = new HashSet<>();
+
+
+        private FileChecker(Jenkins jenkins) throws UnknownHostException {
+            fileSet.add(of("manifest.md", "about", false));
```

Comment:
Here we're creating the checkings to do: the file manifest.md must have the word about not filtered (false param). For every file we set what is expected to be found and whether it should be filtered or not.